src/infra/flow-system/inmem/src/flow/flow_event_store_inmem.rs

```diff
@@ -343,6 +343,34 @@ impl FlowEventStore for FlowEventStoreInMem {
         Box::pin(futures::stream::iter(flow_ids_page))
     }
 
+    #[tracing::instrument(level = "debug", skip_all)]
+    fn get_all_flow_ids_by_datasets(
+        &self,
+        dataset_ids: Vec<DatasetID>,
+        pagination: FlowPaginationOpts,
+    ) -> (usize, FlowIDStream) {
+        let mut total_count = 0;
+        let flow_ids_page: Vec<_> = {
+            let state = self.inner.as_state();
+            let g = state.lock().unwrap();
+            let mut result: Vec<Result<FlowID, _>> = vec![];
+            g.all_flows.iter().rev().for_each(|flow_id| {
+                let flow_key = g.flow_key_by_flow_id.get(flow_id).unwrap();
+                if let FlowKey::Dataset(flow_key_dataset) = flow_key {
+                    if dataset_ids.contains(&flow_key_dataset.dataset_id) {
```

Comment:
Maybe convert  `Vec` to `HashSet` to speedup lookups? It does not have to affect the interface of the repo trait, in the database implementation we only need a vector to generate an SQL `where` clause.

Reply:
Changed